javainterop/nextEntry.go

```diff
@@ -0,0 +1,29 @@
+package javainterop
```

Comment:
Can we put all of this in one `javainterop.go` file that has an alias for every type? We can figure out a script or something to generate the file if that's easier.

Reply:
that has an alias for every type in the ach package right

are we porting every function in the ach package to the javainterop package or just the ones we are trying to reformat (SegmentFile and NextEntry)

Reply:
Yes it should have an alias to every type and function so they're available to someone using Java. We need to offer wrappers for some of the functions that don't compile nicely.

Reply:
Oh alright.

Reply:
@adamdecaf 

I am having a `infinite recursive call` lint issue when calling the `ach` functions in the `javainterop` functions. I don't think this affects the gobind generation but should i be mindful of that?

Reply:
Can you share the full error?

Reply:
`infinite recursive call (SA5007)go-staticcheck`

It is not a full error. Just a warning for the go-staticcheck

Reply:
Are you trying to generate the JNI code for the entire moov-io/ach repository? You should be able to target the `javainterop` package as the only thing to compile/generate.

Reply:
No, I am targeting just the `javainterop` package.

Actually, I realized if we go the route of reference the ach types by embedding them into structs and calling the exported functions on them like you highlighted for the `Iterator` and `NextEntry` below. It eliminates this error. I think it is because of the `alias` thing as it does not necessarily create a newType but just masks the `oldType`

Reply:
Ah that could be. Sounds good!